clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5017,8 +5017,9 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
         EmitNounwindRuntimeCall(Fn);
       }
     }
-
-    EmitUnreachable(Loc);
+    // Do not emit unreachable instruction for Intel NEO driver and Gen "X" GPUs
+    if (getTarget().getTriple().getSubArch() != llvm::Triple::SPIRSubArch_gen)
```

Comment:
I'm not sure that we set `SPIRSubArch_gen` for generic SPIR-V compilation (JIT). Can we disable `noreturn` for all SPIR targets?

Reply:
Should this condition be specific to SYCL devices?  I'm not sure its appropriate to set it for all other offloads, I'm not sure OpenMP or CUDA/etc would necessarily want this change.